temporal-sdk/src/main/java/io/temporal/internal/worker/PollerOptions.java

```diff
@@ -135,8 +137,22 @@ public Builder setPollThreadNamePrefix(String pollThreadNamePrefix) {
 
     public PollerOptions build() {
       if (uncaughtExceptionHandler == null) {
-        uncaughtExceptionHandler = (t, e) -> log.error("uncaught exception", e);
+        uncaughtExceptionHandler =
+            (t, e) -> {
+              if (e instanceof RuntimeException && e.getCause() instanceof StatusRuntimeException) {
+                StatusRuntimeException sre = (StatusRuntimeException) e.getCause();
+                if (sre.getStatus().getCode() == Status.Code.INVALID_ARGUMENT
+                    && sre.getMessage().startsWith("INVALID_ARGUMENT: UnhandledCommand")) {
+                  log.info(
+                      "Failed workflow task caused by signal race condition. This error likely is recoverable.",
```

Comment:
I believe it's it's possible for this to happen for reasons other than just a signal race. It might make more sense to just say "Failed workflow task due to unhandled command. ..."

Reply:
Perhaps also lift this into a static so the test doesn't repeat the message when asserting.

Reply:
Could you please explain what you mean? I'm not sure if it makes sense to make the builder method static, but perhaps you meant something else?

Reply:
I mean lift the string literal up into a constant